lib/corpus.py

```diff
@@ -180,6 +180,12 @@ def all_speakers(self):
         for sc in self.subcorpora:
             yield from sc.all_speakers()
 
+    def top_level_recordings(self):
+        yield from self.recordings
+
+    def top_level_subcorpora(self):
+        yield from self.subcorpora
+
```

Comment:
Suggested implementation for `top_level_speakers()` (see prev comment)
```suggestion

    def top_level_speakers(self):
        yield from self.speakers.values()
```

Only semi related to this PR, but if `self.speakers` is a dict, `yield from self.speakers` would give the keys (speaker name) and not the values (speaker object), as would be necessary, right?
This should be fixed in `all_speakers` as well.

Reply:
Related to this question, why is `self.speakers` a dictionary, if the value also contain the key (`speaker.name`)?

Reply:
Maybe this should be discussed in a new issue and not here

Reply:
`top_level_speakers` should behave the same as `all_speakers`. It's very counter-intuitive that one returns values and the other keys. And as `all_speakers` already exists the one that needs to be changed is `top_level_speakers`.

Reply:
Are you sure that `all_speakers` is not buggy as it is now?

Reply:
I agree that `top_level_speakers` should behave the same as `all_speakers`. However, I think the current way how `all_speakers` works is wrong:

`all_speakers` returns the keys of `self.speakers` which are of type `str`. 
But `Corpus.add_speaker` expects a `Speaker` object (in fact there is an assertion), so the code
```
for speaker in c.all_speakers():
     merged_corpus.add_speaker(speaker)
```
will not work.

I therefore propose to check if `all_speakers` is already used (and if returning the key is expected) and if not change both functions to return the values instead.

Reply:
Looks like the `all_speakers` code was added here: https://github.com/rwth-i6/i6_core/pull/43 And I did not find a use in i6_core or the AppTek repo for it, so it's probably safe to change it.

Reply:
ok, so I probably didn't realize that `self.speakers` is a dict, thus the implementation